ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/router_isis.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+from .utils import UtilsMixin
+
+
+class RouterIsisMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def router_isis(self) -> dict | None:
+        """
+        return structured config for router_isis
+        """
+        if self._underlay_isis is not True:
+            return None
+
+        router_isis = {
+            "instance": self._isis_instance_name,
+            "log_adjacency_changes": True,
+            "net": get(self._hostvars, "switch.isis_net"),
+            "router_id": self._router_id,
+            "is_type": get(self._hostvars, "switch.is_type"),
+            "address_family": ["ipv4 unicast"],
+            "isis_af_defaults": [f"maximum-paths {get(self._hostvars, 'isis_maximum_paths')}"],
+        }
+
+        # no passive interfaces
+        no_passive_interfaces = []
+        for link in self._underlay_links:
+            if link["type"] == "underlay_p2p":
+                no_passive_interfaces.append(link["interface"])
+
```

Comment:
This should be a list comprehension
```suggestion
        no_passive_interfaces = [link["interface"] for link in self._underlay_links if link["type"] == "underlay_p2p"]
```